python/tskit/tables.py

```diff
@@ -2925,7 +2938,13 @@ def keep_intervals(self, intervals, simplify=True, record_provenance=True):
 
     def _check_trim_conditions(self):
         if self.migrations.num_rows > 0:
-            raise ValueError("You cannot trim a tree sequence containing migrations")
+            if (np.min(self.migrations.left) < np.min(self.edges.left)) and (
+                np.max(self.migrations.right) > np.max(self.edges.right)
+            ):
+                raise ValueError(
```

Comment:
What can't we do this? I mean, we'd lose those migrations, right? But should we just issue a warning? Or should we say "this would lose information, so you have to deliberately remove it using delete_intervals([[trim_left, trim_right]]) first"?